### PR TITLE
Accordion update

### PR DESCRIPTION
**Pull request by Swaathik**
## Purpose
Add '*' to the label of an item in Accordion if needed
https://www.ebi.ac.uk/panda/jira/browse/TRM-30429

## Approach
Added a prop to AccordionItem

## Testing
What test(s) did you write to validate and verify your changes?

## Stories
What story(ies) did you write or change to document the functionality / changes?

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why
- [ ] Were all the edited/created files in TypeScript? If not, please list the files you left in JavaScript and the reason to not update them.
- [ ] For the stories you created/updated, are the props modifiables through knobs?

### CODE REVIEW

File: src/components/accordion-search.tsx [hunk 1]
```diff
@@ -83,6 +84,10 @@ type AccordionSearchItemProps = {
    * The user's query, passed to highlight in the item's label
    */
   query: string;
+  /**
+   * Indicates if the item has more values to return on request (in case of xrefs)
```
Comment (dlrice) [inline src/components/accordion-search.tsx:88]: You may be able to be even more generic eg
```
Appends an asterisk to selected items (in the case of xrefs this indicates full xrefs)

File: src/components/accordion-search.tsx [hunk 2]
```diff
@@ -83,6 +84,10 @@ type AccordionSearchItemProps = {
...
+   */
+  addAsterix?: boolean;
```
Comment (dlrice) [inline src/components/accordion-search.tsx:90]:
```suggestion
  addAsterisk?: boolean;
```
I had to google this one as i've never had to spell the word "asterisk"!

File: src/components/accordion-search.tsx [hunk 3]
```diff
@@ -104,6 +110,7 @@ const AccordionSearchCheckbox = ({
         checked={selected.includes(id)}
       />
       <SubstringHighlight substring={query}>{label}</SubstringHighlight>
+      {selected.includes(id) && addAsterix ? '*' : ''}
```
Comment (dlrice) [inline src/components/accordion-search.tsx:113]:
```suggestion
      {addAsterix && selected.includes(id) ? '*' : ''}
```
A minor suggestion. Possibly a little faster to have this order as `addAsterix` will be a very quick quick and so if false it won't bother to look at `selected.includes(id)`

File: src/components/accordion-search.tsx (hunk 2 above)
Comment (Swaathik) [reply src/components/accordion-search.tsx]: My bad. I was so confident this was the correct spelling. Apparently, it is a comic character.

**Review by dlrice — APPROVED** [review]

**Pull request merged**